clients/src/test/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImplTest.java

```diff
@@ -95,6 +98,8 @@ public class MembershipManagerImplTest {
     private BlockingQueue<BackgroundEvent> backgroundEventQueue;
     private BackgroundEventHandler backgroundEventHandler;
     private Time time;
+    private Metrics metrics;
+    private RebalanceMetricsManager rebalanceMetricsManager = mock(RebalanceMetricsManager.class);
```

Comment:
In this class, you sometimes mock the rebalancemetricsmanager, and sometimes you use the real one. I think it's easier to stick to one of the two, so either the metrics manager is part of the unit-under-test or not. You can probably just use the "real one" everywhere?